File: Problems/0441_Arranging_Coins/Arranging_Coins.py
# coding: utf-8

import os
import sys
import time


class Solution:
    def arrangeCoins(self, n):
        """
        :type n: int
        :rtype: int
        """
        row = 1
        while n >= row:
            n = n - row
            row += 1
        return row - 1


def main():
    argv = sys.argv
    argc = len(argv)

    if (argc < 2):
        print("Usage: python %s <testdata.txt>" %(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("%s not found..." %argv[1])
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        print("argv[1] = %s" %temp)
        loop_main(temp)


def loop_main(temp):
    n = int(temp)

    time0 = time.time()

    sl = Solution()
    result = sl.arrangeCoins(n)
    print("result = %d" %result)

    time1 = time.time()

    print("Execute time ... : %f[s]" %(time1 - time0))
    print()


if __name__ == "__main__":
    main()

File: Problems/0441_Arranging_Coins/Arranging_Coins.cs
using System;
using System.Collections.Generic;

public class Solution {

    public int ArrangeCoins(int n)
    {
        if (n == 0)
            return 0;
        var res = (int)(-0.5 + 2 * Math.Sqrt(0.0625 + n / 2));       
        int temp = res + 1;

        if (((temp % 2 == 0) && (n == (temp + 1) * (temp / 2))) ||
            ((temp % 2 == 1) && (n == temp * ((temp + 1) / 2))))
            return temp;
        return res;
    }

    public int ArrangeCoins2(int n)
    {
        int row = 1;

        while (row <= n) {
            n -= row;
            row++;
        }

        return row - 1;
    }

    public void Main(string args)
    {
        int n = int.Parse(args);

        Console.WriteLine("n = " + n.ToString());

        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();
        
        int result = ArrangeCoins(n);

        Console.WriteLine("result = " + result);
        
        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}
